nni/retiarii/nn/pytorch/api.py

```diff
@@ -312,6 +313,125 @@ def __getitem__(self, item):
         return access
 
 
+class Repeat(nn.Module):
+    """
+    Repeat a block by a variable number of times.
+
+    Parameters
+    ----------
+    blocks : function, list of function, module or list of module
+        The block to be repeated. If not a list, it will be replicated into a list. If a function, it will be called to
+        instantiate a module. Otherwise the module will be deep-copied.
+    depth : int or tuple of int
+        If one number, the block will be repeated by a fixed number of times. If a tuple, it should be (min, max),
+        meaning that the block will be repeated at least `min` times and at most `max` times.
+    """
+    def __init__(self,
+                 blocks: Union[Callable[[], nn.Module], List[Callable[[], nn.Module]], nn.Module, List[nn.Module]],
+                 depth: Union[int, Tuple[int, int]], label=None):
+        super().__init__()
+        self._label = label if label is not None else f'valuechoice_{uid()}'
+        self.min_depth = depth if isinstance(depth, int) else depth[0]
+        self.max_depth = depth if isinstance(depth, int) else depth[1]
+        assert self.max_depth >= self.min_depth > 0
+        if not isinstance(blocks, list):
+            if isinstance(blocks, nn.Module):
+                blocks = [blocks] + [copy.deepcopy(blocks) for _ in range(self.max_depth - 1)]
+            else:
+                blocks = [blocks for _ in range(self.max_depth)]
+        assert len(blocks) > 0
+        if not isinstance(blocks[0], nn.Module):
+            blocks = [b() for b in blocks]
+        assert self.max_depth <= len(blocks)
+        self.blocks = nn.ModuleList(blocks)
+
+    @property
+    def label(self):
+        return self._label
+
+    def forward(self, x):
+        for block in self.blocks:
+            x = block(x)
+        return x
+
+
+class Cell(nn.Module):
```

Comment:
I think "cell" could be more general.
op_candidates and num_nodes (or num_ops maybe) are likely to be supported by almost all algorithms, but ops_per_node and merge_op might not.
And from end-user's perspective, without knowledge about the NAS algorithm, they might not understand why there are "ops per node".
My suggestion is, either make optional parameters kwargs, or only support basic params in `Cell` and let algorithms to inherit it.
I do accept current version if it's not considered API freeze.

Reply:
I think the term is widely accepted in NAS literature. This paragraph is taken from a recent literature by Facebook AI.

![image](https://user-images.githubusercontent.com/8463288/119291316-d593c880-bc80-11eb-9f35-3cf0b91bc30c.png)

I did survey a few more papers and they all used the term "cell".

We can think of a better name though.

Reply:
indeed, it is a little confusing that there are two Cells, another one is in our graph ir

Reply:
Better name wanted. Feel free to propose a new one.